pkg/core/blockchain.go

```diff
@@ -1622,7 +1625,7 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 				stop = start + 1
 			}
 			for index := start; index < stop; index++ {
-				err := kvcache.DeleteBlock(bc.GetHeaderHash(index))
+				err := kvcache.DeleteBlock(bc.GetHeaderHash(index), bc.config.Ledger.RemoveUntraceableHeaders)
```

Comment:
Another minor issue is that `getheaders` P2P command relies on fact that header slice is continious, whereas in fact with this setting we need at least to replace `break` with `continue` (or which is even better, check the indexes wrt MaxTraceableBlocks and RemoveUntraceableHeaders setting):
https://github.com/nspcc-dev/neo-go/blob/38f635bce0ff200fde782020128e0db01e394f32/pkg/network/server.go#L1124-L1127

Reply:
A tiny bit better with `continue`, but doesn't change the overall effect --- headers are gone, you can't request them.